### PR TITLE
feat(board): Add Waveshare ESP32-S3-Matrix support

### PR DESCRIPTION
**Pull request by Xylopyrographer**
## Description:

Add support for the Waveshare ESP32-S3-Matrix board.

### CODE REVIEW

**Comment by kdpuvvadi** [comment]
This can be same for ESP32-S3-Zero

**Comment by Xylopyrographer** [comment]
@kdpuvvadi Looks like it should. It has extra pins that won't be picked I by the Arduino "Dx", "Ax" notation.

**Comment by kdpuvvadi** [comment]
Can you point me to right direction or can you do a pr for the same

**Comment by Xylopyrographer** [comment]
@kdpuvvadi 

Looks like support for the ESP32-S3-Zero is already in the Espresssif `arduino-esp32` repository 👍.

So all you need to do is create the board support file and raise a PR to add that to the `pioarduino` repository.

Use the `boards/ws_esp32_s3_matrix.json` as a template and modify it for the Zero. You'll need to open the `boards.txt` file from the Espressif `arduino-esp32` repository to ensure the names in the new  `.json` file you create line up. Use the `boards.txt` entry for the `ESP32-S3-Matrix` as a reference to compare to the  `ws_esp32_s3_matrix.json` file.

To test your `.json` file, copy it into the `boards` folder on your local machine. It should appear in the pio Board Explorer.

File: boards/ws_esp32_s3_matrix.json [hunk 1]
```diff
@@ -0,0 +1,32 @@
+{
+  "build": {
+    "core": "esp32",
+    "f_cpu": "240000000L",
+    "f_flash": "80000000L",
+    "memory_type": "qio_qspi"
```
Comment (kylefmohr) [inline boards/ws_esp32_s3_matrix.json:6]: This needs a comma, it being missing is causing issues: https://github.com/platformio/platformio-core/issues/5055

**Pull request merged**